File: python/7 kyu/282.py
# https://www.codewars.com/kata/598d91785d4ce3ec4f000018/train/python

# Passed

from string import ascii_lowercase

def name_value(my_list):
    letter_to_value = {
        letter: val for val, letter in enumerate(ascii_lowercase, start=1)        
    }
    calc_word_value = lambda word, multiplier: sum(letter_to_value.get(letter, 0) for letter in word) * multiplier
    result = []
    for pos, word in enumerate(my_list, start=1):
        result.append(calc_word_value(word, pos))
        
    return result

output = name_value(["codewars","abc","xyz"])
print(output)

File: nodejs/7 kyu/282.js
// https://www.codewars.com/kata/5a2809dbe1ce0e07f800004d/train/javascript

// Passed

function divisibleByLast(n) {
  let result = [false];
  let digits = String(n).split("");
  for (let idx = 0; idx < digits.length - 1; idx++) {
    let currDigit = Number(digits[idx]);
    let nextDigit = Number(digits[idx + 1]);

    if (currDigit !== 0 && nextDigit % currDigit === 0) {
      result.push(true);
    } else {
      result.push(false);
    }
  }

  return result;
}

const output = divisibleByLast(73312);
console.log(output);